packages/blocks/azure/src/lib/actions/azure-advisor-action.ts

```diff
@@ -0,0 +1,87 @@
+import { createAction, Property } from '@openops/blocks-framework';
+import { azureAuth, dryRunCheckBox } from '@openops/common';
+import { logger } from '@openops/server-shared';
+import { runCommand } from '../azure-cli';
+import { subDropdown, useHostSession } from '../common-properties';
+
+export const advisorAction = createAction({
+  auth: azureAuth,
+  name: 'advisor',
+  description: 'Get Azure Advisor Cost Recommendations',
+  displayName: 'Get Advisor Cost Recommendations',
+  props: {
+    useHostSession: useHostSession,
+    subscriptions: subDropdown,
+    ids: Property.Array({
+      displayName: 'Resource IDs',
+      description:
+        'One or more resource IDs (space-delimited). If provided, no other "Resource Id" arguments should be specified.',
+      required: false,
+    }),
+    resourceGroup: Property.ShortText({
+      displayName: 'Resource Group',
+      description: 'Name of a resource group.',
+      required: false,
+    }),
+    dryRun: dryRunCheckBox(),
+  },
+  async run(context) {
+    const command = buildCommand(context.propsValue);
+    logger.info(`Running command: ${command}`);
+    try {
+      const { dryRun } = context.propsValue;
+
+      if (dryRun) {
+        return 'Step execution skipped, dry run flag enabled. Azure get cost recommendations wont be run.';
+      }
+
+      const result = await runCommand(
+        command,
+        context.auth,
+        context.propsValue.useHostSession?.['useHostSessionCheckbox'],
+        context.propsValue.subscriptions?.['subDropdown'],
+      );
+
+      try {
+        const jsonObject = JSON.parse(result);
+        return jsonObject;
+      } catch (error) {
+        return result;
+      }
+    } catch (error) {
```

Comment:
i know this is very similar to the other action, but as we arent sure if we will keep using the cli in the backend or if we will use the azure API in the future i didnt feel like it makes sense yet to merge the logic, if we add a third action that needs this we can do it then